cpp/src/c_api/graph_sg.cpp

```diff
@@ -292,11 +320,25 @@ struct create_graph_functor : public cugraph::c_api::abstract_functor {
       if (renumber_) {
         *number_map = std::move(new_number_map.value());
       } else {
+        // Ensure vertices are numbered consecutively
+
         number_map->resize(graph->number_of_vertices(), handle_.get_stream());
         cugraph::detail::sequence_fill(handle_.get_stream(),
                                        number_map->data(),
                                        number_map->size(),
                                        graph->view().local_vertex_partition_range_first());
+
+        auto is_consecutive = cugraph::detail::is_equal(
+          handle_.get_stream(),
+          raft::device_span<vertex_t>{vertices.data(), vertices.size()},
+          raft::device_span<vertex_t>{number_map->data(), number_map->size()});
+
+        if (!is_consecutive) {
+          mark_error(
+            CUGRAPH_INVALID_INPUT,
+            "Vertex list must be numbered consecutively from 0 when 'renumber' is 'false'");
+          return;
+        }
```

Comment:
Do you need to keep both `vertices` and `number_map`?

If you just need to keep `number_map`, you can just check `vertices` is an array of consecutive integers starting from 0 and `move` `vertices` to `number_map`.

Reply:
> If you just need to keep number_map

Yes I only need to keep `number_map`.

> you can just check vertices is an array of consecutive integers starting from 0 and move vertices to number_map

That's essentially what the check `cugraph::detail::is_equal` is doing.

Reply:
Yeah... what I meant is that you don't need to allocate memory for `vertices` twice nor call `sequence_fill`.

`thrust::is_equal(vertices.begin(), vertices.end(), thrust::make_counting_iterator(graph_view.local_vertex_partition_range_first());` would be sufficient.

Reply:
Disregard this comment. This code won't be called in normal use cases, so let's just waste little memory/computing here rather than creating another thrust wrapper function.